pls_core/src/inflections/mod.rs

```diff
@@ -106,3 +106,82 @@ where
         Ok(result)
     }
 }
+
+fn get_inflections_for_pattern(
+    pattern: &str,
+    exec_sql: impl Fn(String) -> Result<Vec<Vec<Vec<String>>>, String>,
+) -> Result<Vec<Vec<Vec<String>>>, String> {
+    exec_sql("Select * from {}".replace("{}", pattern))
+}
+
+fn get_words_for_indeclinable_stem(paliword: &str) -> Vec<(String, String, String, String)> {
+    vec![(
+        paliword.chars().filter(|c| !c.is_digit(10)).collect(),
+        paliword.to_string(),
+        " ".to_string(),
+        "ind".to_string(),
+    )]
+}
+
+fn get_words_for_irregular_stem(
+    paliword: &str,
+    pattern: &str,
+    _exec_sql: fn(&str) -> Result<String, String>,
+) -> Vec<(String, String, String, String)> {
```

Comment:
this is called primitive obsession code smell.

please define a struct, with properly named fields and pass it around.

optional comment - so feel free to not take it in.